tutorials/Multi_objective_multi_fidelity_BO.ipynb

```diff
@@ -0,0 +1,1720 @@
+    "MC_SAMPLES = 128  # Number of Monte Carlo samples\n",
+    "\n",
+    "standard_boundsMO = torch.tensor(\n",
+    "    [[0.0] * dim_xMO, [1.0] * dim_xMO], **tkwargs\n",
+    ")  # Bounds for MO only optimization\n",
+    "standard_bounds = torch.tensor(\n",
+    "    [[0.0] * dim_x, [1.0] * dim_x], **tkwargs\n",
+    ")  # Bounds for MOMF optimization\n",
+    "verbose = True"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Problem Setup ###\n",
+    "The problem as described before is a modified multi-fidelity version of Branin-Currin (BC) function that results in a 3 x 2 problem. A simple fidelity objective is also defined here which is a linear function of the input fidelity. We also design a wrapper function around the BC that takes care of interfacing torch with numpy and appends the fidelity objective with the BC functions.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def fid_obj(X):\n",
+    "    # A Fidelity Objective that can be thought of as a trust objective. Higher Fidelity simulations are rewarded as being more\n",
+    "    # trustworthy. Here we consider just a linear fidelity objective.\n",
+    "    fid_obj = 1 * X[:, -1]\n",
+    "    return fid_obj\n",
+    "\n",
+    "\n",
+    "def problem(x, dim_y):\n",
+    "    # Wrapper around the Objective function to take care of fid_obj stacking\n",
+    "    y = BC(x)  # The Branin-Currin is called\n",
+    "    fid = fid_obj(x)  # Getting the fidelity objective values\n",
+    "    fid_out = torch.reshape(fid, (-1, 1))\n",
+    "    y_out = torch.cat(\n",
+    "        [y, fid_out], 1\n",
+    "    )  # Concatenating objective values with fid_objective\n",
+    "    return y_out"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Model Initialization ###\n",
+    "We use a multi-output SingleTaskGP to model the problem with a homoskedastic Gaussian likelihood with an inferred noise level. \n",
+    "The model is initialized with 5 random points where the fidelity dimension of the initial points is sampled from a distribution : $p(s)=C(s)^{-1}$ "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from botorch.models.gp_regression import SingleTaskGP\n",
+    "from botorch.models.transforms.outcome import Standardize\n",
+    "from gpytorch.mlls.exact_marginal_log_likelihood import ExactMarginalLogLikelihood\n",
+    "\n",
+    "exp_arg = torch.tensor(4)\n",
+    "\n",
+    "\n",
+    "def gen_init_data(dim_x, points, dim_y):\n",
+    "    # Generates training data with Fidelity dimension sampled from a probability distribution that depends on Cost function\n",
+    "    train_x = torch.rand(points, dim_x)\n",
+    "    fid_samples = torch.linspace(0, 1, 101)\n",
+    "    prob = 1 / cost_func(\n",
```

Comment:
Can you also add a comment describing the procedure for generating initial fidelities?